nixos/modules/hardware/network/intel-2200bg.nix

```diff
@@ -23,7 +23,7 @@
 
   config = lib.mkIf config.networking.enableIntel2200BGFirmware {
 
-    hardware.enableAllFirmware = true;
+    hardware.firmware = [ pkgs.intel2200BGFirmware ];
```

Comment:
Generally `hardware.firmware` should be a sorted set on which each module appends to. For now let's just add it to `enableAllFirmware`, otherwise the meaning will be "all firmware except ipw2200 :)